### PR TITLE
v2: Only split when inline callbacks consume some bytes

### PR DESCRIPTION
**Pull request by Ambrevar**
The former hacks around maybeLineBreak and Smartypants are no longer
needed.
The algorithm has been streamlined: shorter, simpler, faster.
The 'currBlock' field of the parser is gone.

Bonus: A rough perf increase of 6-8%!

### CODE REVIEW

File: inline.go [hunk 1]
```diff
@@ -189,35 +167,36 @@ func codeSpan(p *parser, data []byte, offset int) int {
...
+	return 0, nil
 }
 
 // newline without two spaces works when HardLineBreak is enabled
-func lineBreak(p *parser, data []byte, offset int) int {
+func lineBreak(p *parser, data []byte, offset int) (int, *Node) {
 	if p.flags&HardLineBreak != 0 {
-		p.currBlock.AppendChild(NewNode(Hardbreak))
-		return 1
+		// log.Print("LINEBREAK")
```
Comment (rtfb) [inline inline.go]: nit: debug log

File: inline.go [hunk 2]
```diff
@@ -189,35 +167,36 @@ func codeSpan(p *parser, data []byte, offset int) int {
...
+		return 1, NewNode(Hardbreak)
 	}
-	return 0
+	// log.Print("NOLINEBREAK")
```
Comment (rtfb) [inline inline.go]: one more

**Comment by rtfb** [comment]
This is brilliant! Thanks a lot!

Please fix the couple stray debug logs and let's merge it.

File: inline.go (hunk 2 above)
Comment (Ambrevar) [inline inline.go]: That's odd, I recall removing those 2 lines...

**Comment by Ambrevar** [comment]
Done.
Please squash-merge.

**Comment by rtfb** [comment]
Thanks again! :-)

**Pull request merged**